sw/device/tests/dif/dif_pinmux_test.cc

```diff
@@ -0,0 +1,379 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "gtest/gtest.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/testing/mock_mmio.h"
+
+extern "C" {
+#include "pinmux_regs.h"  // Generated.
+#include "sw/device/lib/dif/dif_pinmux.h"
```

Comment:
Should we be inserting `extern "C"` in our headers I wonder?

Reply:
I think so. Wrapping includes in extern "C" is kind of rude IMO; it's probably good practice to mark polyglot headers as such with the usual `__cplusplus` incantation. I'll file an issue, but I don't think we should worry about it in this change.

https://github.com/lowRISC/opentitan/issues/1934

Reply:
We still have to wrap "pinmux_regs.h" as they are not auto generated with
```
#ifdef __cplusplus
extern "C" {
#endif  // __cplusplus 
```

So I am leaving this as it is for now.

Reply:
Sounds good, We need to look at the fall-out from the polyglot headers rules anyway. :)

Reply:
#2308 is ready for the generated headers, but I don't know when it will be merged. I'm happy for this to be left as-is and cleared up when we make all the DIF headers polyglot (which we probably want to do all at once).